api/rosetta/data_integration_test.go

```diff
@@ -97,6 +111,68 @@ func setupAPI(t *testing.T, db *badger.DB) *rosetta.Data {
 	return controller
 }
 
+func setupRecorder(endpoint string, input interface{}, options ...func(*http.Request)) (*httptest.ResponseRecorder, echo.Context, error) {
+
+	var payload []byte
+
+	switch input.(type) {
+
+	case []byte:
+		payload = input.([]byte)
+
+	default:
+		enc, err := json.Marshal(input)
+		if err != nil {
+			return nil, echo.New().AcquireContext(), fmt.Errorf("could not encode input: %w", err)
+		}
+
+		payload = enc
+	}
+
+	req := httptest.NewRequest(http.MethodPost, endpoint, bytes.NewReader(payload))
+	req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+	for _, opt := range options {
+		opt(req)
+	}
+
+	rec := httptest.NewRecorder()
+
+	ctx := echo.New().NewContext(req, rec)
+
+	return rec, ctx, nil
+}
+
+func checkRosettaError(statusCode int, def meta.ErrorDefinition, desc string, details map[string]interface{}) func(t assert.TestingT, err error, v ...interface{}) bool {
```

Comment:
The description and details parameters here should definitely be removed. They are 100% just for human consumption and we shouldn't check them, just like we shouldn't check the exact string printing of an error.